File: notifications/src/main/java/io/castled/notifications/trigger/StringConditionEvaluator.java
package io.castled.notifications.trigger;

import io.castled.notifications.exceptions.CastledRuntimeException;
import io.castled.notifications.trigger.models.PropertyOperation;
import io.castled.notifications.trigger.models.SingleValueOperation;
import io.castled.notifications.trigger.models.BetweenOperation;

public class StringConditionEvaluator implements ParamsConditionEvaluator {

    @Override
    public boolean evaluateCondition(Object value, PropertyOperation propertyOperation) {
        String textValue = (String) value;
        switch (propertyOperation.getOperationType()) {
            case EQ:
                return ((SingleValueOperation) propertyOperation).getValue().equals(textValue);
            case NEQ:
                if (textValue == null) {
                    return true;
                }
                return !((SingleValueOperation) propertyOperation).getValue().equals(textValue);
            case GT:
                return textValue!= null && ((SingleValueOperation) propertyOperation).getValue().compareTo(textValue) < 0;
            case LT:
                return textValue!= null && ((SingleValueOperation) propertyOperation).getValue().compareTo(textValue) > 0;
            case GTE:
                return textValue!= null && ((SingleValueOperation) propertyOperation).getValue().compareTo(textValue) <= 0;
            case LTE:
                return textValue!= null && ((SingleValueOperation) propertyOperation).getValue().compareTo(textValue) >= 0;
            case BETWEEN:
                BetweenOperation doubleValueOperation = ((BetweenOperation) propertyOperation);
                return textValue!= null && doubleValueOperation.getFrom().compareTo(textValue) < 0 && doubleValueOperation.getTo().compareTo(textValue) > 0;
            default:
                throw new CastledRuntimeException(String.format("Operations type %s not supported for numeric operand", propertyOperation.getPropertyType()));
        }
    }
}

File: notifications/src/main/java/io/castled/notifications/trigger/StringConditionEvaluator.kt
package io.castled.notifications.trigger

import io.castled.notifications.trigger.enums.OperationType
import io.castled.notifications.trigger.models.PropertyOperation
import io.castled.notifications.trigger.models.SingleValueOperation

internal class StringConditionEvaluator : ParamsConditionEvaluator() {

    override fun evaluateCondition(value: Any?, propertyOperation: PropertyOperation): Boolean {
        val textValue = value as String? ?: return false
        return when (propertyOperation.operationType) {
            OperationType.EQ -> (propertyOperation as SingleValueOperation).value == textValue
            OperationType.NEQ -> (propertyOperation as SingleValueOperation).value != textValue
            else -> {
                logger.error("Operations type ${propertyOperation.propertyType} not supported for string operand")
                return false
            }
        }
    }
}